### ISSUE TITLE
colcon version-check fails to execute because of "too many requests by the client"

### ISSUE THREAD

**Issue by wmmc88** [colcon/colcon-core#437]
This is on a fresh install of colcon on Ubuntu 20.04. I installed following the following instructions:
```
sudo sh -c 'echo "deb [arch=amd64,arm64] http://repo.ros2.org/ubuntu/main `lsb_release -cs` main" > etc/apt/sources.list.d/ros2-latest.list'
curl -s https://raw.githubusercontent.com/ros/rosdistro/master/ros.asc | sudo apt-key add -
sudo apt update
sudo apt install python3-colcon-common-extensions
```

`apt show python3-colcon*` gives the following output:
```
Package: python3-colcon-argcomplete
Version: 0.3.3-1
Priority: optional
Section: python
Maintainer: Dirk Thomas <web@dirk-thomas.net>
Installed-Size: 39.9 kB
Depends: python3-argcomplete, python3:any (>= 3.5~), python3-colcon-core
Download-Size: 5,452 B
APT-Manual-Installed: no
APT-Sources: http://packages.ros.org/ros/ubuntu focal/main amd64 Packages
Description: Completion for colcon command lines using argcomplete.
 colcon-argcomplete
 ==================
 .
 An extension for `colcon-core <https://github.com/colcon/colcon-core>`_ to provide command line completion using `argcomplete <https://github.com/kislyuk/argcomplete>`_.

Package: python3-colcon-core
Version: 0.6.1-2
Priority: optional
Section: python
Maintainer: Dirk Thomas <web@dirk-thomas.net>
Installed-Size: 404 kB
Depends: python3-distlib, python3-empy, python3-pkg-resources, python3-pytest, python3-pytest-cov, python3:any (>= 3.5~), python3-setuptools
Homepage: https://colcon.readthedocs.io
Download-Size: 59.6 kB
APT-Manual-Installed: no
APT-Sources: http://packages.ros.org/ros/ubuntu focal/main amd64 Packages
Description: Command line tool to build sets of software packages.
 colcon - collective construction
 ================================
 .
 ``colcon`` is a command line tool to improve the workflow of building, testing and using multiple software packages.
 It automates the process, handles the ordering and sets up the environment to use the packages.
 .
 For more information see `colcon.readthedocs.io <https://colcon.readthedocs.io>`_.

Package: python3-colcon-bash
Version: 0.4.2-1
Priority: optional
Section: python
Maintainer: Dirk Thomas <web@dirk-thomas.net>
Installed-Size: 37.9 kB
Depends: python3:any (>= 3.5~), python3-colcon-core (>= 0.3.18)
Download-Size: 5,356 B
APT-Manual-Installed: no
APT-Sources: http://packages.ros.org/ros/ubuntu focal/main amd64 Packages
Description: Extension for colcon to provide Bash scripts.
 colcon-bash
 ===========
 .
 An extension for `colcon-core <https://github.com/colcon/colcon-core>`_ to provide `Bash <https://www.gnu.org/software/bash/>`_ scripts.

Package: python3-colcon-cd
Version: 0.1.1-1
Priority: optional
Section: python
Maintainer: Dirk Thomas <web@dirk-thomas.net>
Installed-Size: 29.7 kB
Depends: python3:any (>= 3.5~), python3-colcon-core (>= 0.4.1), python3-colcon-package-information
Download-Size: 3,784 B
APT-Manual-Installed: no
APT-Sources: http://packages.ros.org/ros/ubuntu focal/main amd64 Packages
Description: A shell function for colcon to change the current working di
 colcon-cd
 =========
 .
 A shell function for `colcon-core <https://github.com/colcon/colcon-core>`_ to change the current working directory.

Package: python3-colcon-package-information
Version: 0.3.3-1
Priority: optional
Section: python
Maintainer: Dirk Thomas <web@dirk-thomas.net>
Installed-Size: 54.3 kB
Depends: python3:any (>= 3.5~), python3-colcon-core (>= 0.5.2)
Download-Size: 8,340 B
APT-Manual-Installed: no
APT-Sources: http://packages.ros.org/ros/ubuntu focal/main amd64 Packages
Description: Extension for colcon to output package information.
 colcon-package-information
 ==========================
 .
 An extension for `colcon-core <https://github.com/colcon/colcon-core>`_ to provide information about the packages.

Package: python3-colcon-cmake
Version: 0.2.26-2
Priority: optional
Section: python
Maintainer: Dirk Thomas <web@dirk-thomas.net>
Installed-Size: 88.1 kB
Depends: python3:any (>= 3.5~), cmake, python3-colcon-core (>= 0.5.6), python3-colcon-library-path, python3-colcon-test-result (>= 0.3.3)
Homepage: https://colcon.readthedocs.io
Download-Size: 14.9 kB
APT-Manual-Installed: no
APT-Sources: http://packages.ros.org/ros/ubuntu focal/main amd64 Packages
Description: Extension for colcon to support CMake packages.
 colcon-cmake
 ============
 .
 An extension for `colcon-core <https://github.com/colcon/colcon-core>`_ to support `CMake <https://cmake.org>`_ projects.

Package: python3-colcon-library-path
Version: 0.2.1-1
Priority: optional
Section: python
Maintainer: Dirk Thomas <web@dirk-thomas.net>
Installed-Size: 26.6 kB
Depends: python3:any (>= 3.5~), python3-colcon-core
Homepage: https://colcon.readthedocs.io
Download-Size: 3,528 B
APT-Manual-Installed: no
APT-Sources: http://packages.ros.org/ros/ubuntu focal/main amd64 Packages
Description: Extension for colcon adding an environment variable to find
 colcon-library-path
 ===================
 .
 An extension for `colcon-core <https://github.com/colcon/colcon-core>`_ to set an environment variable to find shared libraries at runtime.

Package: python3-colcon-test-result
Version: 0.3.8-1
Priority: optional
Section: python
Maintainer: Dirk Thomas <web@dirk-thomas.net>
Installed-Size: 44.0 kB
Depends: python3:any (>= 3.5~), python3-colcon-core
Download-Size: 7,144 B
APT-Manual-Installed: no
APT-Sources: http://packages.ros.org/ros/ubuntu focal/main amd64 Packages
Description: Extension for colcon to provide information about the test r
 colcon-test-result
 ==================
 .
 An extension for `colcon-core <https://github.com/colcon/colcon-core>`_ to provide information about the test results.

Package: python3-colcon-common-extensions
Version: 0.2.1-1
Priority: optional
Section: python
Maintainer: Dirk Thomas <web@dirk-thomas.net>
Installed-Size: 21.5 kB
Depends: python3:any (>= 3.5~), python3-colcon-argcomplete, python3-colcon-bash, python3-colcon-cd, python3-colcon-cmake, python3-colcon-core, python3-colcon-defaults, python3-colcon-devtools, python3-colcon-library-path, python3-colcon-metadata, python3-colcon-notification, python3-colcon-output, python3-colcon-package-information, python3-colcon-package-selection, python3-colcon-parallel-executor, python3-colcon-powershell, python3-colcon-python-setup-py, python3-colcon-recursive-crawl, python3-colcon-ros, python3-colcon-test-result, python3-colcon-zsh
Download-Size: 2,880 B
APT-Manual-Installed: yes
APT-Sources: http://packages.ros.org/ros/ubuntu focal/main amd64 Packages
Description: Meta package aggregating colcon-core and common extensions.
 colcon-common-extensions
 ========================
 .
 A meta package aggregating `colcon-core <https://github.com/colcon/colcon-core>`_ as well as a set of common extensions.

Package: python3-colcon-defaults
Version: 0.2.5-1
Priority: optional
Section: python
Maintainer: Dirk Thomas <web@dirk-thomas.net>
Installed-Size: 32.8 kB
Depends: python3-yaml, python3:any (>= 3.5~), python3-colcon-core (>= 0.5.5)
Download-Size: 5,232 B
APT-Manual-Installed: no
APT-Sources: http://packages.ros.org/ros/ubuntu focal/main amd64 Packages
Description: Extension for colcon to read defaults from a config file.
 colcon-defaults
 ===============
 .
 An extension for `colcon-core <https://github.com/colcon/colcon-core>`_ to provide custom default values for the command line arguments from a configuration file.
 .
 For more information about the format of the configuration file see `colcon.readthedocs.io <http://colcon.readthedocs.io/en/released/user/configuration.html#defaults-yaml>`_.

Package: python3-colcon-devtools
Version: 0.2.2-1
Priority: optional
Section: python
Maintainer: Dirk Thomas <web@dirk-thomas.net>
Installed-Size: 33.8 kB
Depends: python3:any (>= 3.5~), python3-colcon-core
Download-Size: 4,448 B
APT-Manual-Installed: no
APT-Sources: http://packages.ros.org/ros/ubuntu focal/main amd64 Packages
Description: Extension for colcon to provide information about all extens
 colcon-devtools
 ===============
 .
 An extension for `colcon-core <https://github.com/colcon/colcon-core>`_ to provide information about the plugin system.

Package: python3-colcon-metadata
Version: 0.2.5-1
Priority: optional
Section: python
Maintainer: Dirk Thomas <web@dirk-thomas.net>
Installed-Size: 62.5 kB
Depends: python3-yaml, python3:any (>= 3.5~), python3-colcon-core
Download-Size: 9,172 B
APT-Manual-Installed: no
APT-Sources: http://packages.ros.org/ros/ubuntu focal/main amd64 Packages
Description: Extension for colcon to read package metadata from files.
 colcon-metadata
 ===============
 .
 An extension for `colcon-core <https://github.com/colcon/colcon-core>`_ to fetch and manage package metadata from repositories.
 .
 For an example repository containing metadata see `colcon-metadata-repository <https://github.com/colcon/colcon-metadata-repository>`_.

Package: python3-colcon-notification
Version: 0.2.13-1
Priority: optional
Section: python
Maintainer: Dirk Thomas <web@dirk-thomas.net>
Installed-Size: 99.3 kB
Depends: python3:any (>= 3.5~), python3-colcon-core (>= 0.3.7), python3-notify2
Download-Size: 10.5 kB
APT-Manual-Installed: no
APT-Sources: http://packages.ros.org/ros/ubuntu focal/main amd64 Packages
Description: Extension for colcon to provide status notifications.
 colcon-notification
 ===================
 .
 An extension for `colcon-core <https://github.com/colcon/colcon-core>`_ to provide status notifications.

Package: python3-colcon-output
Version: 0.2.12-2
Priority: optional
Section: python
Maintainer: Dirk Thomas <web@dirk-thomas.net>
Installed-Size: 44.0 kB
Depends: python3:any (>= 3.5~), python3-colcon-core (>= 0.3.8)
Homepage: https://colcon.readthedocs.io
Download-Size: 6,500 B
APT-Manual-Installed: no
APT-Sources: http://packages.ros.org/ros/ubuntu focal/main amd64 Packages
Description: Extension for colcon to customize the output in various ways
 colcon-output
 =============
 .
 An extension for `colcon-core <https://github.com/colcon/colcon-core>`_ to customize the output in various ways.

Package: python3-colcon-package-selection
Version: 0.2.10-2
Priority: optional
Section: python
Maintainer: Dirk Thomas <web@dirk-thomas.net>
Installed-Size: 60.4 kB
Depends: python3:any (>= 3.5~), python3-colcon-core (>= 0.3.19)
Homepage: https://colcon.readthedocs.io
Download-Size: 7,536 B
APT-Manual-Installed: no
APT-Sources: http://packages.ros.org/ros/ubuntu focal/main amd64 Packages
Description: Extension for colcon to select the packages to process.
 colcon-package-selection
 ========================
 .
 An extension for `colcon-core <https://github.com/colcon/colcon-core>`_ to select a subset of packages for processing.

Package: python3-colcon-parallel-executor
Version: 0.2.4-1
Priority: optional
Section: python
Maintainer: Dirk Thomas <web@dirk-thomas.net>
Installed-Size: 32.8 kB
Depends: python3:any (>= 3.5~), python3-colcon-core (>= 0.3.15)
Download-Size: 5,200 B
APT-Manual-Installed: no
APT-Sources: http://packages.ros.org/ros/ubuntu focal/main amd64 Packages
Description: Extension for colcon to process packages in parallel.
 colcon-parallel-executor
 ========================
 .
 An extension for `colcon-core <https://github.com/colcon/colcon-core>`_ to process packages in parallel.

Package: python3-colcon-powershell
Version: 0.3.6-1
Priority: optional
Section: python
Maintainer: Dirk Thomas <web@dirk-thomas.net>
Installed-Size: 42.0 kB
Depends: python3:any (>= 3.5~), python3-colcon-core (>= 0.4.2)
Download-Size: 6,520 B
APT-Manual-Installed: no
APT-Sources: http://packages.ros.org/ros/ubuntu focal/main amd64 Packages
Description: Extension for colcon to provide PowerShell scripts.
 colcon-powershell
 =================
 .
 An extension for `colcon-core <https://github.com/colcon/colcon-core>`_ to provide `PowerShell <https://github.com/PowerShell/PowerShell>`_ scripts.

Package: python3-colcon-python-setup-py
Version: 0.2.7-2
Priority: optional
Section: python
Maintainer: Dirk Thomas <web@dirk-thomas.net>
Installed-Size: 37.9 kB
Depends: python3-pkg-resources, python3:any (>= 3.5~), python3-colcon-core (>= 0.3.10), python3-setuptools
Homepage: https://colcon.readthedocs.io
Download-Size: 6,244 B
APT-Manual-Installed: no
APT-Sources: http://packages.ros.org/ros/ubuntu focal/main amd64 Packages
Description: Extension for colcon to support Python packages with the met
 colcon-python-setup-py
 ======================
 .
 An extension for `colcon-core <https://github.com/colcon/colcon-core>`_ to identify packages with a ``setup.py`` file by introspecting the arguments to the ``setup()`` function call of  `setuptools <https://setuptools.readthedocs.io/en/latest/>`_.

Package: python3-colcon-recursive-crawl
Version: 0.2.1-1
Priority: optional
Section: python
Maintainer: Dirk Thomas <web@dirk-thomas.net>
Installed-Size: 26.6 kB
Depends: python3:any (>= 3.5~), python3-colcon-core (>= 0.5.0)
Download-Size: 3,648 B
APT-Manual-Installed: no
APT-Sources: http://packages.ros.org/ros/ubuntu focal/main amd64 Packages
Description: Extension for colcon to recursively crawl for packages.
 colcon-recursive-crawl
 ======================
 .
 An extension for `colcon-core <https://github.com/colcon/colcon-core>`_ to recursively crawl for packages.

Package: python3-colcon-ros
Version: 0.3.21-2
Priority: optional
Section: python
Maintainer: Dirk Thomas <web@dirk-thomas.net>
Installed-Size: 85.0 kB
Depends: python3:any (>= 3.5~), python3-catkin-pkg-modules (>= 0.4.14), python3-colcon-cmake (>= 0.2.6), python3-colcon-core (>= 0.5.3), python3-colcon-pkg-config, python3-colcon-python-setup-py (>= 0.2.4), python3-colcon-recursive-crawl
Homepage: https://colcon.readthedocs.io
Download-Size: 11.4 kB
APT-Manual-Installed: no
APT-Sources: http://packages.ros.org/ros/ubuntu focal/main amd64 Packages
Description: Extension for colcon to support ROS packages.
 colcon-ros
 ==========
 .
 An extension for `colcon-core <https://github.com/colcon/colcon-core>`_ to support `ROS packages <http://www.ros.org>`_.

Package: python3-colcon-zsh
Version: 0.4.0-1
Priority: optional
Section: python
Maintainer: Dirk Thomas <web@dirk-thomas.net>
Installed-Size: 35.8 kB
Depends: python3:any (>= 3.5~), python3-colcon-core (>= 0.3.18)
Download-Size: 5,316 B
APT-Manual-Installed: no
APT-Sources: http://packages.ros.org/ros/ubuntu focal/main amd64 Packages
Description: Extension for colcon to provide Z shell scripts.
 colcon-zsh
 ==========
 .
 An extension for `colcon-core <https://github.com/colcon/colcon-core>`_ to provide `Z shell <http://www.zsh.org>`_ scripts.

Package: python3-colcon-coveragepy-result
Version: 0.0.8-1
Priority: optional
Section: python
Maintainer: Christophe Bedard <bedard.christophe@gmail.com>
Installed-Size: 42.0 kB
Depends: python3-coverage, python3:any (>= 3.5~), python3-colcon-core
Download-Size: 6,464 B
APT-Sources: http://packages.ros.org/ros/ubuntu focal/main amd64 Packages
Description: colcon extension for collecting coverage.py results
 # colcon-coveragepy-result
 .
 [![PyPI](https://img.shields.io/pypi/v/colcon-coveragepy-result)](https://pypi.org/project/colcon-coveragepy-result/)
 [![GitHub Action Status](https://github.com/colcon/colcon-coveragepy-result/workflows/Test/badge.svg)](https://github.com/colcon/colcon-coveragepy-result/actions)
 .
 A [colcon](https://colcon.readthedocs.io/) extension for collecting [coverage.py](https://coverage.readthedocs.io/en/stable/) results.
 .
 ## Usage
 .
 1. Build packages with coverage, e.g. using a [colcon mixin](https://colcon.readthedocs.io/en/released/reference/verb/mixin.html) for [coverage](https://github.com/colcon/colcon-mixin-repository/blob/master/coverage.mixin)
    ```shell
    $ colcon build --mixin coverage-pytest
    ```
 1. Test packages with coverage, again using a mixin
    ```shell
    $ colcon test --mixin coverage-pytest
    ```
 1. Collect coverage results
    ```shell
    $ colcon coveragepy-result

Package: python3-colcon-ed
Version: 0.1.2-2
Priority: optional
Section: python
Maintainer: Chen Bainian <chenbn@artc.a-star.edu.sg>
Installed-Size: 31.7 kB
Depends: python3:any (>= 3.5~), python3-colcon-core (>= 0.5.2)
Homepage: https://colcon.readthedocs.io
Download-Size: 4,844 B
APT-Sources: http://packages.ros.org/ros/ubuntu focal/main amd64 Packages
Description: Extension for colcon to edit file in directories.
 colcon-ed
 =========
 .
 An extension for `colcon-core <https://github.com/colcon/colcon-core>`_ to edit a file within a package.
 .
 See the `documentation <https://colcon.readthedocs.io/en/master/reference/verb/edit.html>`_ on how to use it.

Package: python3-colcon-lcov-result
Version: 0.5.0-2
Priority: optional
Section: python
Maintainer: Juan Pablo Samper <jp.samper@apex.ai>
Installed-Size: 49.2 kB
Depends: python3:any (>= 3.5~), python3-colcon-core
Homepage: https://colcon.readthedocs.io
Download-Size: 8,324 B
APT-Sources: http://packages.ros.org/ros/ubuntu focal/main amd64 Packages
Description: Extension for colcon to gather test results.
 colcon-lcov-result
 ==================
 .
 An extension for `colcon-core <https://github.com/colcon/colcon-core>`_ to provide aggregate
 coverage results using `LCOV <http://ltp.sourceforge.net/coverage/lcov.php>`_.
 .
 LCOV is a graphical front-end for GCC's coverage testing tool
 `gcov <https://gcc.gnu.org/onlinedocs/gcc/Gcov.html>`_, producing the following
 coverage metrics:
 .
 - Statement coverage
 - Function coverage
 - Branch coverage
 .
 For more information, see `this paper
 <http://ltp.sourceforge.net/documentation/technical_papers/gcov-ols2003.pdf>`_
 and `this Wikipedia page <https://en.wikipedia.org/wiki/Code_coverage>`_.
 .
 .
 Usage

Package: python3-colcon-mixin
Version: 0.2.0-1
Priority: optional
Section: python
Maintainer: Dirk Thomas <web@dirk-thomas.net>
Installed-Size: 66.6 kB
Depends: python3-yaml, python3:any (>= 3.5~), python3-colcon-core (>= 0.5.6)
Download-Size: 10.4 kB
APT-Sources: http://packages.ros.org/ros/ubuntu focal/main amd64 Packages
Description: Extension for colcon to read CLI mixins from files.
 colcon-mixin
 ============
 .
 An extension for `colcon-core <https://github.com/colcon/colcon-core>`_ to fetch and manage CLI mixins from repositories.
 .
 For an example repository containing mixins see `colcon-mixin-repository <https://github.com/colcon/colcon-mixin-repository>`_.

Package: python3-colcon-pkg-config
Version: 0.1.0-1
Priority: optional
Section: python
Maintainer: Dirk Thomas <web@dirk-thomas.net>
Installed-Size: 25.6 kB
Depends: python3:any (>= 3.5~), python3-colcon-core
Homepage: https://colcon.readthedocs.io
Download-Size: 3,208 B
APT-Manual-Installed: no
APT-Sources: http://packages.ros.org/ros/ubuntu focal/main amd64 Packages
Description: Extension for colcon adding an environment variable to find
 colcon-pkg-config
 ===================
 .
 An extension for `colcon-core <https://github.com/colcon/colcon-core>`_ to set an environment variable to find pkg-config files.
```

Error Log:
```
❯ colcon version-check
colcon-argcomplete 0.3.3: up-to-date
[0.483s] ERROR:colcon:colcon version-check: <Fault -32500: 'HTTPTooManyRequests: The action could not be performed because there were too many requests by the client. Limit may reset in 1 seconds.'>
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/colcon_core/command.py", line 528, in verb_main
    rc = context.args.main(context=context)
  File "/usr/lib/python3/dist-packages/colcon_devtools/verb/version_check.py", line 36, in main
    versions = pypi.package_releases(dist.project_name)
  File "/usr/lib/python3.8/xmlrpc/client.py", line 1109, in __call__
    return self.__send(self.__name, args)
  File "/usr/lib/python3.8/xmlrpc/client.py", line 1450, in __request
    response = self.__transport.request(
  File "/usr/lib/python3.8/xmlrpc/client.py", line 1153, in request
    return self.single_request(host, handler, request_body, verbose)
  File "/usr/lib/python3.8/xmlrpc/client.py", line 1169, in single_request
    return self.parse_response(resp)
  File "/usr/lib/python3.8/xmlrpc/client.py", line 1341, in parse_response
    return u.close()
  File "/usr/lib/python3.8/xmlrpc/client.py", line 655, in close
    raise Fault(**self._stack[0])
xmlrpc.client.Fault: <Fault -32500: 'HTTPTooManyRequests: The action could not be performed because there were too many requests by the client. Limit may reset in 1 seconds.'>
```

**Comment by christophebedard** [comment]
There is an open PR to fix this (by switching to another method for checking/querying versions): https://github.com/colcon/colcon-devtools/pull/20

**Comment by wmmc88** [comment]
Looks like this was resolved in #20 and released in colcon-devtools 0.2.3! I just tested it out and it works! thanks!
